planner/core/exhaust_physical_plans.go

```diff
@@ -1940,16 +1940,18 @@ func (p *LogicalJoin) exhaustPhysicalPlans(prop *property.PhysicalProperty) ([]P
 	joins := make([]PhysicalPlan, 0, 8)
 	canPushToTiFlash := p.canPushToCop(kv.TiFlash)
 	if p.ctx.GetSessionVars().IsMPPAllowed() && canPushToTiFlash {
-		if p.shouldUseMPPBCJ() {
-			mppJoins := p.tryToGetMppHashJoin(prop, true)
-			if (p.preferJoinType & preferBCJoin) > 0 {
-				return mppJoins, true, nil
-			}
-			joins = append(joins, mppJoins...)
-		} else {
-			mppJoins := p.tryToGetMppHashJoin(prop, false)
-			joins = append(joins, mppJoins...)
+		shuffleJoins := p.tryToGetMppHashJoin(prop, false)
+		if (p.preferJoinType&preferShuffleJoin) > 0 && len(shuffleJoins) > 0 { // has shuffle_join hints
+			return shuffleJoins, true, nil
+		}
```

Comment:
If we have both BCJoin hint and shuffle join hint, it seems we will chose the shuffle join. Is it expected?

Reply:
The behavior of using multiple conflict hints together seems undefined. Let's notify our users on docs and any expectation wouldn't be guaranteed here.